SDK: Improve @dotcms/angular Documentation

### Task description
As a developer using the `@dotcms/angular` library, I want the documentation to be clear and easy to follow, so that I can easily understand how to use the library in my projects.

### Acceptance Criteria
1. The README file should provide a clear introduction to the library and its purpose.
2. The README should include full step by step on how to use the library.
3. Update the README of the `example/angular`, include env variables, tokens, everything needed to start the example
4. Comprehensive JSDoc comments on all the components inputs, properties, classes, methods etc...
6. Any dependencies or prerequisites needed to use the library should be clearly stated.

### Problem
For example the library have this:

```
Add the dotcms config in the Angular app ApplicationConfig

export const appConfig: ApplicationConfig = {
    providers: [
        provideDotcmsClient(DOTCMS_CLIENT_CONFIG),
        provideRouter(routes),
    ],
};
```
 But is not explaining where `provideDotcmsClient` is coming from, or how to create or anything, is missing to much context.


### Assumptions & Initiation Needs
N/A

The writing is clear, with only a couple of small snags:

> - A set of Angular components developer for dotCMS page rendering and editor integration. 
- developer -> "developed"

> ## Configutarion 
- "Configuration"

(under "Client Usage")
> For more information to how to use DotCms Client
- "For more information on how to use the dotCMS Client,"

Otherwise, the copy itself reads well.

As far as testing the code samples myself, I'm unable to even install the library due to dependency tree issues. Any suggestions on this?

```
jamie@Jamies-MacBook-Air ~ % npm install @dotcms/angular
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: undefined@undefined
npm error Found: @angular/common@17.3.12
npm error node_modules/@angular/common
npm error   peer @angular/common@"^17.1.0" from @dotcms/angular@0.0.1-alpha.38
npm error   node_modules/@dotcms/angular
npm error     @dotcms/angular@"*" from the root project
npm error   peer @angular/common@"17.3.12" from @angular/router@17.3.12
npm error   node_modules/@angular/router
npm error     peer @angular/router@"^17.1.0" from @dotcms/angular@0.0.1-alpha.38
npm error     node_modules/@dotcms/angular
npm error       @dotcms/angular@"*" from the root project
npm error   2 more (@angular/platform-browser, @tinymce/tinymce-angular)
npm error
npm error Could not resolve dependency:
npm error peer @angular/common@"18.2.6" from @angular/forms@18.2.6
npm error node_modules/@angular/forms
npm error   peer @angular/forms@">=16.0.0" from @tinymce/tinymce-angular@8.0.1
npm error   node_modules/@tinymce/tinymce-angular
npm error     peer @tinymce/tinymce-angular@"^8.0.0" from @dotcms/angular@0.0.1-alpha.38
npm error     node_modules/@dotcms/angular
npm error       @dotcms/angular@"*" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
```

--- a/core-web/libs/sdk/angular/README.md
+++ b/core-web/libs/sdk/angular/README.md
@@ -1,6 +1,6 @@
 # @dotcms/angular
 
-`@dotcms/angular` is the official Angular library designed to work seamlessly with dotCMS. This library simplifies the process of rendering dotCMS pages and integrating with the [Universal Visual Editor](dotcms.com/docs/latest/universal-visual-editor) in your Angular applications.
+`@dotcms/angular` is the official Angular library designed to work seamlessly with dotCMS. This library simplifies the process of rendering dotCMS pages and integrating with the [Universal Visual Editor](https://www.dotcms.com/docs/latest/universal-visual-editor) in your Angular applications.
 
 ## Table of Contents
 
@@ -37,7 +37,7 @@ Or using Yarn:
 yarn add @dotcms/angular
 ```
 
-## Configutarion
+## Configuration
 ### Provider Setup
 We need to provide the information of our dotCMS instance
 
@@ -51,10 +51,11 @@ const DOTCMS_CLIENT_CONFIG: ClientConfig = {
     siteId: environment.siteId
 };
 ```
-And add this config in the Angular app ApplicationConfig.
+
+Add this configuration to `ApplicationConfig` in your Angular app.
 
 `src/app/app.config.ts`
-```javascript
+```typescript
 import { InjectionToken } from '@angular/core';
 import { ClientConfig, DotCmsClient } from '@dotcms/client';
 
@@ -70,36 +71,40 @@ export const appConfig: ApplicationConfig = {
     ],
 };
 ```
+
 This way, we will have access to `DOTCMS_CLIENT_TOKEN` from anywhere in our application.
 
 ### Client Usage
 To interact with the client and obtain information from, for example, our pages
 
-```javascript
-private readonly client = inject(DOTCMS_CLIENT_TOKEN);
+```typescript
+export class YourComponent {
+    private readonly client = inject(DOTCMS_CLIENT_TOKEN);
 
-this.client.page
-    .get({ ...pageParams })
-    .then((response) => {
-        // Use your response
-    })
-    .catch((e) => {
-      const error: PageError = {
-        message: e.message,
-        status: e.status,
-      };
-      // Use the error response
-    })
+    this.client.page
+        .get({ ...pageParams })
+        .then((response) => {
+            // Use your response
+        })
+        .catch((e) => {
+            const error: PageError = {
+                message: e.message,
+                status: e.status,
+            };
+            // Use the error response
+        })
+}
 ```
-For more information on how to use the dotCMS Client,you can visit the [documentation](https://github.com/dotCMS/core/blob/main/core-web/libs/sdk/client/README.md)
+
+For more information on how to use the dotCMS Client, you can visit the [documentation](https://www.github.com/dotCMS/core/blob/main/core-web/libs/sdk/client/README.md)
 
 ## DotCMS Page API
 
 The `DotcmsLayoutComponent` requires a `DotCMSPageAsset` object to be passed in to it. This object represents a dotCMS page and can be fetched using the `@dotcms/client` library.
 
-- [DotCMS Official Angular Example](https://github.com/dotCMS/core/tree/main/examples/angular)
+- [DotCMS Official Angular Example](https://www.github.com/dotCMS/core/tree/main/examples/angular)
 - [`@dotcms/client` documentation](https://www.npmjs.com/package/@dotcms/client)
-- [Page API documentation](https://dotcms.com/docs/latest/page-api)
+- [Page API documentation](https://www.dotcms.com/docs/latest/page-api)
 
 ## Components
 
@@ -167,17 +172,58 @@ This setup allows for dynamic rendering of different content types on your dotCM
 
 ## Troubleshooting
 
-If you encounter issues:
+If you encounter issues while using `@dotcms/angular`, here are some common problems and their solutions:
 
-1. Ensure all dependencies are correctly installed and up to date.
-2. Verify that your dotCMS configuration (URL, auth token, site ID) is correct.
-3. Check the browser console for any error messages.
-4. Refer to the [dotCMS documentation](https://dotcms.com/docs/) for additional guidance.
+1. **Dependency Issues**:
+   - Ensure that all dependencies, such as `@dotcms/client`, `@angular/core`, and `rxjs`, are correctly installed and up-to-date. You can verify installed versions by running:
+     ```bash
+     npm list @dotcms/client @angular/core rxjs
+     ```
+   - If there are any missing or incompatible versions, reinstall dependencies by running:
+     ```bash
+     npm install
+     ```
+     or
+     ```bash
+     npm install --force
+     ```
+
+2. **Configuration Errors**:
+   - **DotCMS Configuration**: Double-check that your `DOTCMS_CLIENT_CONFIG` settings (URL, auth token, site ID) are correct and aligned with the environment variables. For example:
+     ```typescript
+     const DOTCMS_CLIENT_CONFIG: ClientConfig = {
+         dotcmsUrl: environment.dotcmsUrl,  // Ensure this is a valid URL
+         authToken: environment.authToken,  // Ensure auth token has the correct permissions
+         siteId: environment.siteId         // Ensure site ID is valid and accessible
+     };
+     ```
+   - **Injection Issues**: Ensure that `DOTCMS_CLIENT_TOKEN` is provided globally. Errors like `NullInjectorError` usually mean the token hasn’t been properly added to the `ApplicationConfig`. Verify by checking `src/app/app.config.ts`.
+
+3. **Network and API Errors**:
+   - **dotCMS API Connectivity**: If API calls are failing, check your browser’s Network tab to ensure requests to `dotcmsUrl` are successful. For CORS-related issues, ensure that your dotCMS server allows requests from your application’s domain.
+   - **Auth Token Permissions**: If you’re seeing `401 Unauthorized` errors, make sure the auth token used in `DOTCMS_CLIENT_CONFIG` has appropriate permissions in dotCMS for accessing pages and content.
+
+4. **Page and Component Rendering**:
+   - **Dynamic Imports**: If you’re encountering issues with lazy-loaded components, make sure dynamic imports are correctly set up, as in:
+     ```typescript
+      const DYNAMIC_COMPONENTS: DotCMSPageComponent = {
+        Activity: import('../pages/content-types/activity/activity.component').then(
+            (c) => c.ActivityComponent
+        )
+      };
+     ```
+   - **Invalid Page Assets**: Ensure that `pageAsset` objects are correctly formatted. Missing fields in `pageAsset` can cause errors in `DotcmsLayoutComponent`. Validate the structure by logging `pageAsset` before passing it in.
+
+5. **Common Angular Errors**:
+   - **Change Detection**: Angular sometimes fails to detect changes with dynamic content. If `DotcmsLayoutComponent` isn’t updating as expected, you may need to call `ChangeDetectorRef.detectChanges()` manually.
+   - **TypeScript Type Errors**: Ensure all types (e.g., `DotCMSPageAsset`, `DotCMSPageComponent`) are imported correctly from `@dotcms/angular`. Type mismatches can often be resolved by verifying imports.
+
+6. **Consult Documentation**: Refer to the official [dotCMS documentation](https://dotcms.com/docs/) and the [@dotcms/angular GitHub repository](https://github.com/dotCMS/core). These sources often provide updates and additional usage examples.
 
 ## Contributing
 
-GitHub pull requests are the preferred method to contribute code to dotCMS. Before any pull requests can be accepted, an automated tool will ask you to agree to the [dotCMS Contributor's Agreement](https://gist.github.com/wezell/85ef45298c48494b90d92755b583acb3).
+GitHub pull requests are the preferred method to contribute code to dotCMS. Before any pull requests can be accepted, an automated tool will ask you to agree to the [dotCMS Contributor's Agreement](https://www.gist.github.com/wezell/85ef45298c48494b90d92755b583acb3).
 
 ## Licensing
 
-dotCMS comes in multiple editions and as such is dual licensed. The dotCMS Community Edition is licensed under the GPL 3.0 and is freely available for download, customization and deployment for use within organizations of all stripes. dotCMS Enterprise Editions (EE) adds a number of enterprise features and is available via a supported, indemnified commercial license from dotCMS. For the differences between the editions, see [the feature page](http://dotcms.com/cms-platform/features).
+dotCMS comes in multiple editions and as such is dual licensed. The dotCMS Community Edition is licensed under the GPL 3.0 and is freely available for download, customization and deployment for use within organizations of all stripes. dotCMS Enterprise Editions (EE) adds a number of enterprise features and is available via a supported, indemnified commercial license from dotCMS. For the differences between the editions, see [the feature page](http://www.dotcms.com/cms-platform/features).

